### PR TITLE
Update postgresql conf

### PR DESCRIPTION
**Pull request by michaellzc**
This PR copied the updated pg-conf from https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/792

Also, we use `Files.Get` template function to reference the static file to keep configmap template clean. My eyes will hurt less next time I have to update the postgresql.conf :)

### Checklist

- [x] Follow the [manual testing process](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/TEST.md)

### Test plan

```sh
helm template sourcegraph charts/sourcegraph/. > bundle.yaml
```

all configmaps look good

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #105** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/105" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/105?utm_source=stack-comment).

File: charts/sourcegraph/templates/codeinsights-db/codeinsights-db.ConfigMap.yaml [hunk 1]
```diff
@@ -13,756 +13,6 @@ metadata:
...
-
-    #------------------------------------------------------------------------------
-    # CUSTOMIZED OPTIONS
-    #------------------------------------------------------------------------------
-
-    # Add settings for extensions here
-
-    {{ tpl .Values.codeInsightsDB.additionalConfig . | nindent 4 | trim }}
+    {{- .Files.Get "files/codeinsights-db/conf/postgresql.conf" | nindent 4 }}
+    {{- tpl .Values.codeInsightsDB.additionalConfig . | nindent 4 | trim }}
```
Comment (michaellzc) [inline charts/sourcegraph/templates/codeinsights-db/codeinsights-db.ConfigMap.yaml:17]: nit: `-`- remove the extra new line above

File: charts/sourcegraph/files/codeinsights-db/conf/postgresql.conf [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+autovacuum_naptime = 10
+shared_preload_libraries = ''
+max_locks_per_transaction = 64
+
+
+#------------------------------------------------------------------------------
+# CUSTOMIZED OPTIONS
+#------------------------------------------------------------------------------
+
+# Add your customization here
```
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/files/codeinsights-db/conf/postgresql.conf:56]: Is this instruction (and the two other references above) confusing now in helm? Customers won't be adding the customizations directly there, but by setting the additionalConfig value.

**Review by caugustus-sourcegraph — APPROVED** [review]
One small concern about the info text but functionally looks good

**Pull request merged**